File: src/main/scala/io/digitalmagic/example3/PlaceOrderExample.scala
package io.digitalmagic.example3

import OrderApi._

object PlaceOrderExample {

  def main(args:Array[String]) = {

    process (
      Order buy 100 shares_Of "Apple" at_Limit_Price 12.54 value_As Standard all_Or_None,
      Order sell 50 shares_Of "IBM" at_Limit_Price 11.45 value_As Premium
    )

  }

}

File: src/main/java/io/digitalmagic/example2/PlaceOrderExample.java
package io.digitalmagic.example2;

import io.digitalmagic.OrderProcessor;
import io.digitalmagic.PremiumOrderValuer;
import io.digitalmagic.StandardOrderValuer;

public class PlaceOrderExample {

    private static void process(Order ... orders) {
        for (Order order: orders) {
            OrderProcessor.process(order);
        }
    }

    public static void main(String[] args) {
        process(
                Order.buy(100)
                        .sharesOf("Apple")
                        .atLimitPrice(12.54)
                        .allOrNone()
                        .valueAs(new StandardOrderValuer())
                        .build(),

            Order.sell(50)
                    .sharesOf("IBM")
                    .atLimitPrice(11.45)
                    // We don't specify allOrNone - because wan't allow partial order
                    .valueAs(new PremiumOrderValuer())
                    .build()
        );
    }
}
